core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -226,17 +226,14 @@ func (r *EvmRegistry) GetActiveUpkeepIDsByType(ctx context.Context, triggers ...
 	return keys, nil
 }
 
-// TODO: should be called with ocr2keepers.UpkeepPayload
-func (r *EvmRegistry) CheckUpkeep(ctx context.Context, mercuryEnabled bool, keys ...ocr2keepers.UpkeepKey) ([]ocr2keepers.UpkeepResult, error) {
+func (r *EvmRegistry) CheckUpkeeps(ctx context.Context, mercuryEnabled bool, keys ...ocr2keepers.UpkeepPayload) ([]ocr2keepers.CheckResult, error) {
```

Comment:
last week, Awbrey and I talked about this and our conclusion is that we don't need this boolean here.

we can probably use ConfiguredUpkeep to pass this bool or even remove this completely.